### PR TITLE
ref(test): Convert datascrubbing edit modal to rtl

### PR DESCRIPTION
**Pull request by priscilawebdev**

### CODE REVIEW

File: static/app/views/settings/components/dataScrubbing/modals/form/index.tsx [hunk 1]
```diff
@@ -150,6 +147,7 @@ class Form extends Component<Props<Values, KeysOfUnion<Values>>, State> {
                 onChange={this.handleChange('pattern')}
                 value={values.pattern}
                 onBlur={onValidate('pattern')}
+                aria-label={t('Regular expression')}
```
Comment (evanpurkhiser) [inline static/app/views/settings/components/dataScrubbing/modals/form/index.tsx:150]: wanna fix the ID thing since it's open again haha

**Review by evanpurkhiser — APPROVED** [review]

File: static/app/views/settings/components/dataScrubbing/modals/form/index.tsx (hunk 1 above)
Comment (evanpurkhiser) [reply static/app/views/settings/components/dataScrubbing/modals/form/index.tsx:150]: Did you put it on both the control and the field?

**Review by k-fish — APPROVED** [review]
🎉

File: static/app/views/settings/components/dataScrubbing/modals/form/index.tsx (hunk 1 above)
Comment (priscilawebdev) [reply static/app/views/settings/components/dataScrubbing/modals/form/index.tsx]: oh I see... I updated the code. thanks!

File: static/app/views/settings/components/dataScrubbing/modals/form/index.tsx [hunk 2]
```diff
@@ -133,10 +131,10 @@ class Form extends Component<Props<Values, KeysOfUnion<Values>>, State> {
           </Field>
           {values.type === RuleType.PATTERN && (
             <Field
-              data-test-id="regex-field"
               label={t('Regex matches')}
               help={t('Custom regular expression (see documentation)')}
               inline={false}
+              id={t('Regular expression')}
```
Comment (evanpurkhiser) [inline static/app/views/settings/components/dataScrubbing/modals/form/index.tsx:137]: wait this should be an id string hahah, not a translation

Comment (priscilawebdev) [reply static/app/views/settings/components/dataScrubbing/modals/form/index.tsx:137]: damn!

Comment (evanpurkhiser) [reply static/app/views/settings/components/dataScrubbing/modals/form/index.tsx:137]: like `id="regexMatches"` or whatever

Comment (priscilawebdev) [reply static/app/views/settings/components/dataScrubbing/modals/form/index.tsx]: thanks. Updated 😄

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**